src/matchers/types.ts

```diff
@@ -31,26 +38,8 @@ declare global {
 }
 
 // Explicit `@jest/globals` `expect` matchers.
+// @ts-ignore
```

Comment:
Wouldn't this effectively force all your users to set `skipLibCheck: true` in tsconfig? 😔

Reply:
Do you have ideas how to resolve any other way?
<img width="1011" alt="image" src="https://github.com/callstack/react-native-testing-library/assets/6368606/c9e1febb-c2a2-4509-9862-61ed41be08d4">

Reply:
idk when testing against your repro repo the mentioned `@ts-ignore` seems to work fine. Seems like it might just be suppring check for `@jest/expect` existence which does not prevent it from using in test app with `@jest/expect`.

Note: In order to trigger the `@jest/globals` module typing you need to use `expect` from `@jest/globals` and not a global one:

<img width="986" alt="image" src="https://github.com/callstack/react-native-testing-library/assets/6368606/9b1c60f1-c71a-4785-b289-aaa15d72fedf">

Which BTW triggers a lot other TS errors when invoking `tsc` but not from RNTL itself, but things like:
* node_modules/@types/node/buffer.d.ts
* node_modules/react-native/types/modules/globals.d.ts